### PR TITLE
Replaces Lame Captain Laser With Badass Gigarevolver

### PR DESCRIPTION
**Pull request by TheDanseMacabre**
This PR replaces the captain's antique laser (a LRP relic by all accounts and functionally a bit underwhelming/nonsensical) with a child object of the autorevolver, which has slightly worse stats compared to its parent but is still very lethal. It is specifically fluffed as a largely ceremonial/decorative revolver popular with high-ranking SCC personnel. Sprite by Kyres.

### CODE REVIEW

**Comment by TheDanseMacabre** [comment]
!review

**Comment by alsoandanswer** [comment]
Hard no from me. This isn't much of an improvement to our current sprite, and changing it to a ballistic weapon seems like a pretty odd and arbritrary reason.

**Comment by TheDanseMacabre** [comment]
> Hard no from me. This isn't much of an improvement to our current sprite, and changing it to a ballistic weapon seems like a pretty odd and arbritrary reason.

While I'm sorry you don't like the sprite, the captain's antique laser is an extremely ugly and outdated sprite for 2010~, with a nonsensical meme description and pitiful ingame performance. The odd and arbitrary reason is pretty straightforward: this is supposed to give the captain an impressive and powerful sidearm with limited uses. It is supposed to be an intimidating weapon for the highest ranking character on the ship, and achieves this. The fact it is a ballistic weapon is irrelevant.

**Comment by Generalcamo** [comment]
> This isn't much of an improvement to our current sprite
Can you please elaborate on this, so that the sprite creator may make any needed modifications?

**Comment by alsoandanswer** [comment]
For the sprite being old, a better alternative would've been just to port the /tg/ sprite rather than go through all this trouble.

As for the weapon performance, there could've just been a buff and tweak.

As for it being changed for a ballistic weapon, I'd still prefer it remaining as an energy weapon.

File: code/game/objects/structures/displaycase.dm [hunk 1]
```diff
@@ -113,4 +113,7 @@
 			health_check()
 
 /obj/structure/displaycase/captain_laser
-	spawn_contained_type = /obj/item/gun/energy/captain
\ No newline at end of file
+	spawn_contained_type = /obj/item/gun/energy/captain
+
+/obj/structure/displaycase/captain_revolver
+	spawn_contained_type = /obj/item/gun/projectile/revolver/mateba/captain_revolver
```
Comment (SleepyGemmy) [inline code/game/objects/structures/displaycase.dm:119]:
```suggestion
	spawn_contained_type = /obj/item/gun/projectile/revolver/mateba/captain
```

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 2]
```diff
@@ -52,6 +52,19 @@
 	ammo_type = /obj/item/ammo_casing/a454
 	magazine_type = /obj/item/ammo_magazine/a454
 
+/obj/item/gun/projectile/revolver/mateba/captain_revolver
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/projectile/revolver.dm:55]:
```suggestion
/obj/item/gun/projectile/revolver/mateba/captain
```

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 3]
```diff
@@ -52,6 +52,19 @@
...
+	name = "SCC command revolver"
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/projectile/revolver.dm:56]:
```suggestion
	name = "\improper SCC command revolver"
```

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 4]
```diff
@@ -52,6 +52,19 @@
...
+	desc = "A ludicrously powerful .454 revolver with equally ludicrous recoil which is issued by the Conglomerate to the administrators of critical facilities and vessels. While revolvers may be a thing of the past, the stopping power displayed by this weapon is second to none."
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/projectile/revolver.dm:57]:
```suggestion
	desc = "A ludicrously powerful .454 revolver with equally ludicrous recoil which is issued by the SCC to the administrators of critical facilities and vessels. While revolvers may be a thing of the past, the stopping power displayed by this weapon is second to none."
```

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 5]
```diff
@@ -52,6 +52,19 @@
...
+	desc_extended = "A Zavodskoi Interstellar design from the mid-2450s intended for export to the Eridani Corporate Federation and Republic of Biesel, the Protektor \
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/projectile/revolver.dm:58]:
```suggestion
	desc_extended = "A Zavodskoi Interstellar design from the mid 2450s intended for export to the Eridani Corporate Federation and the Republic of Biesel, the Protektor \
```

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 6]
```diff
@@ -52,6 +52,19 @@
...
+	Revolver was never designed with practicality in mind. The .454 rounds fired from this weapon are liable to snap the wrist of an unprepared shooter and \
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/projectile/revolver.dm:59]:
```suggestion
	revolver was never designed with practicality in mind. The .454 rounds fired from this weapon are liable to snap the wrist of an unprepared shooter and \
```

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 7]
```diff
@@ -52,6 +52,19 @@
...
+	any following shots will be difficult to place onto a human-sized target due to the recoil, let alone a skrell. But nobody buys a Protektor for the purpose of \
+	practicality: they buy it due to having too much money and wanting a revolver large enough for their ego. Isn't that why you're holding it right now, captain?"
```
Comment (SleepyGemmy) [inline code/modules/projectiles/guns/projectile/revolver.dm:61]:
```suggestion
	practicality: they buy it due to having too much money and wanting a revolver large enough for their ego."
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Comment by TheDanseMacabre** [comment]
I don't really think a /tg/ port is an improvement by any means.

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 8]
```diff
@@ -52,6 +52,39 @@
 	ammo_type = /obj/item/ammo_casing/a454
 	magazine_type = /obj/item/ammo_magazine/a454
 
+/obj/item/gun/projectile/revolver/mateba/captain
+	name = "\improper SCC command autorevolver"
+	desc = "A ludicrously powerful .454 autorevolver with equally ludicrous recoil which is issued by the SCC to the administrators of critical facilities and vessels. While revolvers may be a thing of the past, the stopping power displayed by this weapon is second to none."
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/revolver.dm:57]: I'm not an english expect by any means, but the "which is issued" part feels like it should be preceded by a comma, or the "which" be a "that"

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 9]
```diff
@@ -52,6 +52,39 @@
...
+	desc_info = "In order to accurately fire this revolver, it must be wielded. Additionally, if you fire this revolver unwielded and you are not a G2 or Unathi, you will drop it."
+	desc_extended = "A Zavodskoi Interstellar design from the mid 2450s intended for export to the Eridani Corporate Federation and the Republic of Biesel, the Protektor \
+	revolver was never designed with practicality in mind. The .454 rounds fired from this weapon are liable to snap the wrist of an unprepared shooter and \
+	any following shots will be difficult to place onto a human-sized target due to the recoil, let alone a skrell. But nobody buys a Protektor for the purpose of \
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/revolver.dm:61]: The "But nobody" part could sound better if it was after a comma, or the "but" a "however,"

**Review by FluffyGhoster — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]
Fine with it, but the ammo needs to remain unattainable.

**Comment by TheDanseMacabre** [comment]
> Fine with it, but the ammo needs to remain unattainable.

No problem with that on my end.

**Comment by alsoandanswer** [comment]
said in discord, reiterating here there should be a feature to choose whether you want a ballistic or the old laser weapon like how polaris allows you to choose your weapon.

**Comment by NonQueueingMatt** [comment]
> said in discord, reiterating here there should be a feature to choose whether you want a ballistic or the old laser weapon like how polaris allows you to choose your weapon.

Unwanted feature at the moment.

**Pull request merged**